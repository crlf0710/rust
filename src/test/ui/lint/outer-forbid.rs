// Forbidding a group (here, `unused`) overrules subsequent allowance of both
// the group, and an individual lint in the group (here, `unused_variables`);
// and, forbidding an individual lint (here, `non_snake_case`) overrules
// subsequent allowance of a lint group containing it (here, `nonstandard_style`). See
// Issue #42873.

#![forbid(unused, non_snake_case)]

#[allow(unused_variables)] //~ WARNING overruled
fn foo() {} //~ ERROR function is never used: `foo`

#[allow(unused)] //~ WARNING overruled
fn bar() {} //~ ERROR function is never used: `bar`

#[allow(nonstandard_style)] //~ WARNING overruled
fn main() {
    println!("hello forbidden world")
}
