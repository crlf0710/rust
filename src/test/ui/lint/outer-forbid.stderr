warning[E0453]: allow(unused_variables) overruled by outer forbid(unused)
  --> $DIR/outer-forbid.rs:9:9
   |
LL | #![forbid(unused, non_snake_case)]
   |           ------ `forbid` level set here
LL | 
LL | #[allow(unused_variables)]
   |         ^^^^^^^^^^^^^^^^ overruled by previous forbid

warning[E0453]: allow(unused) overruled by outer forbid(unused)
  --> $DIR/outer-forbid.rs:12:9
   |
LL | #![forbid(unused, non_snake_case)]
   |           ------ `forbid` level set here
...
LL | #[allow(unused)]
   |         ^^^^^^ overruled by previous forbid

warning[E0453]: allow(nonstandard_style) overruled by outer forbid(non_snake_case)
  --> $DIR/outer-forbid.rs:15:9
   |
LL | #![forbid(unused, non_snake_case)]
   |                   -------------- `forbid` level set here
...
LL | #[allow(nonstandard_style)]
   |         ^^^^^^^^^^^^^^^^^ overruled by previous forbid

error: function is never used: `foo`
  --> $DIR/outer-forbid.rs:10:4
   |
LL | fn foo() {}
   |    ^^^
   |
note: the lint level is defined here
  --> $DIR/outer-forbid.rs:7:11
   |
LL | #![forbid(unused, non_snake_case)]
   |           ^^^^^^
   = note: `#[forbid(dead_code)]` implied by `#[forbid(unused)]`

error: function is never used: `bar`
  --> $DIR/outer-forbid.rs:13:4
   |
LL | fn bar() {}
   |    ^^^

error: aborting due to 2 previous errors; 3 warnings emitted

For more information about this error, try `rustc --explain E0453`.
